File: 06302023/03_07_Fix_The_String_Sum/python/fix_the_string_sum.py
# https://www.codewars.com/kata/5848cd33c3689be0dc00175c
# Level 7 kyu

# Help Johnny!
# He can't make his code work!
# Johnny is trying to make a function that encodes 2 strings into their ASCII values and sum them together, 
# but he can't find the error in his code! Help him!

# Function
def add(string_one: str, string_two: str) -> int:
    """Converts arguments to ASCII values and sums them together."""
    string_one: bytes = string_one.encode()
    string_two: bytes = string_two.encode()
    return sum(string_one + string_two)

# Tests
# test.assert_equals(add('a', 'b'), 195)
# test.assert_equals(add('aa', 'bb'), 390)

File: 06302023/03_07_Fix_The_String_Sum/java/FixTheStringSum.java
public class FixTheStringSum {
    public static void main(String[] args) {
        System.out.println(add("aa", "bb"));
    }

    public static int add(String stringOne, String stringTwo) {
        int stringOneSum = 0;
        int stringTwoSum = 0;

        for (int i = 0; i < stringOne.length(); i++) {
            stringOneSum += (int) stringOne.charAt(i);
        }
        for (int i = 0; i < stringTwo.length(); i++) {
            stringTwoSum += (int) stringTwo.charAt(i);
        }
        return stringOneSum + stringTwoSum;
    }
}
